templates/ovncontroller/bin/functions

```diff
@@ -52,31 +52,59 @@ function configure_external_ids {
 # Configure bridge mappings and physical bridges
 function configure_physical_networks {
     local OvnBridgeMappings=""
+    local br_new=""
     for physicalNetwork in ${PhysicalNetworks}; do
         br_name="br-${physicalNetwork}"
-        ovs-vsctl --may-exist add-br ${br_name}
-        ovs-vsctl --may-exist add-port ${br_name} ${physicalNetwork}
         bridgeMapping="${physicalNetwork}:${br_name}"
         if [ -z "$OvnBridgeMappings"]; then
             OvnBridgeMappings=$bridgeMapping
+            br_new=$br_name
         else
             OvnBridgeMappings="${OvnBridgeMappings},${bridgeMapping}"
+            br_new="${br_new} ${br_name}"
+        fi
+    done
+
+    # Current configured bridges.
+    ovn_bms=$(ovs-vsctl --if-exists get open . external_ids:ovn-bridge-mappings|tr -d '"')
+    local br_current=""
+    for bm in ${ovn_bms/,/ }; do
+        if [ -z "$br_current"]; then
+            br_current=${bm##*:}
+        else
+            br_current="${br_current} ${bm##*:}"
         fi
     done
-    if [ -n "$OvnBridgeMappings" ]; then
-        ovs-vsctl set open . external-ids:ovn-bridge-mappings=${OvnBridgeMappings}
-    else
-        # If NicMappings not defined or cleared, let's ensure bridge and bridge-mappings are removed
-        ovn_bms=$(ovs-vsctl --if-exists get open . external_ids:ovn-bridge-mappings|tr -d '"')
-        for bm in ${ovn_bms/,/ }; do
-            ovs-vsctl --if-exists del-br ${bm##*:}
-        done
-        ovs-vsctl --if-exists remove open . external_ids ovn-bridge-mappings
 
-        # If NicMappings not defined or cleared, let's ensure patch ports are removed
-        patch_ports=$(ovs-vsctl --columns=name --bare find interface type=patch)
-        for port in ${patch_ports}; do
-            ovs-vsctl --if-exists del-port br-int $port
+    # Bridges to add and delete.
+    local br_to_delete="$(comm -23 <(sort <(echo $br_current | xargs -n1)) <(sort <(echo $br_new | xargs -n1)))"
```

Comment:
these lines are duplicative and not easy to parse. Could you please move the common parts into a function with a comprehensible name? E.g. you could have a function called `get_unique` that receives two lists and returns the difference; then a function called `sort_list` that receives a string and returns an array of strings (sorted), then you could:

```
local br_to_delete=$(get_unique $(sort_list $br_current $br_new))
local br_to_add=$(get_unique $(sort_list $br_new $br_current))
```